shell/shared/shell.js

```diff
@@ -154,6 +166,49 @@ if (Meteor.isServer) {
     self.ready();
   });
 
+  Meteor.publish("myNotYetCompleteReferrals", function() {
+    if (!this.userId) {
+      return [];
+    }
+
+    // A not-yet-complete referral shows up as an Identity where referredBy=this.userId.
+    var notCompletedReferralIdentities = Meteor.users.find(
+      {referredBy: this.userId,
+       "profile.name": {$exists: true}},
+      {fields: {
+        "_id": 1,
+        "referredBy": 1,
+        "profile.name": 1}});
+    return notCompletedReferralIdentities;
+  });
+
+  Meteor.publish("myReferrals", function() {
+    if (!this.userId) {
```

Comment:
Question for @jparyani - is this more idiomatic as a `check(this.userId, String)`?